### DISCUSSION TITLE
How to install shield for only one module in codeigniter 4 project?

### DISCUSSION THREAD

**Discussion by nguyennamit** [codeigniter4/shield#905] [Q&A]
I am building a project using codeigniter 4, it has 2 modules: Admin (this is the section for administrators) and Guest (this is the section for users). I want to use Shield to run only the Admin module, is it possible? and how do I install it?

**Comment by datamweb** [comment]
edit **Routes.php :**

```php
$routes->group('admin', ['filter'=> 'session'], static function ($routes) {
    
// e.g 
$routes->get('panel', 'AdminPanel::index', ['filter'=> 'group:admin']);

service('auth')->routes($routes);
      
});
```
see http://localhost:8080/admin/panel

**Reply by nguyennamit** [reply]
But when I run another module, the shield files still load, and I don't want that because it slows down the page loading speed. Is there a way to make all Shield features run only when I call the Admin Module?

**Reply by kenjis** [reply]
Don't use `service('auth')` or `auth()` or anything in Shield at all when you run another module.

1. If you use `service('auth')->routes($routes)` in `app/Config/Routes.php`, the Shield files are loaded.
2. Don't use the `Auth` collector in Debugbar. There is no way to disable only the `Auth` collector. So change to the production mode.
3. Update Shiled to the latest `develop` branch (read [Upgrade Guide](https://github.com/codeigniter4/shield/blob/develop/UPGRADING.md#upgrade-guide)), and remove `'auth'` in $helpers in app/Config/Autoload.php. And load the auth helper manually when you need.

**Reply by kenjis** [reply]
I sent a PR #908
